xds/internal/xdsclient/xdsresource/listener_resource_type.go

```diff
@@ -143,39 +143,42 @@ func (l *ListenerResourceData) Raw() *anypb.Any {
 // ListenerWatcher wraps the callbacks to be invoked for different
 // events corresponding to the listener resource being watched.
 type ListenerWatcher interface {
-	// OnUpdate is invoked to report an update for the resource being watched.
-	OnUpdate(*ListenerResourceData, OnDoneFunc)
-
-	// OnError is invoked under different error conditions including but not
+	// OnResourceChanged is invoked to notify the watcher of a new version of
+	// the resource received from the xDS server or an error indicating the
+	// reason why the resource cannot be obtained.
+	//
+	// It is invoked under different error conditions including but not
 	// limited to the following:
-	//	- authority mentioned in the resource is not found
-	//	- resource name parsing error
-	//	- resource deserialization error
-	//	- resource validation error
-	//	- ADS stream failure
-	//	- connection failure
-	OnError(error, OnDoneFunc)
-
-	// OnResourceDoesNotExist is invoked for a specific error condition where
-	// the requested resource is not found on the xDS management server.
-	OnResourceDoesNotExist(OnDoneFunc)
+	//      - authority mentioned in the resource is not found
+	//      - resource name parsing error
+	//      - resource validation error (if resource is not cached)
+	//      - ADS stream failure (if resource is not cached)
+	//      - connection failure (if resource is not cached)
+	OnResourceChanged(*ResourceDataOrError, OnDoneFunc)
+
+	// If resource is already cached, it is invoked under different error
```

Comment:
Please start with the docstring with the name of the function, as is customary in Go.